### PR TITLE
feat: add /capacity/pool and /capacity/pool/product endpoints

### LINKED ISSUE #119: Staking Data Implementation (Part 1)
## Task

add 2 new endpoints:

* `/capacity/pools/:poolId` - Returns the current capacity for all products in a specific pool
* `/capacity/pools/:poolId/products/:productId` - Returns the current capacity for a specific product in a specific pool

### PR DESCRIPTION
**Pull request by rackstar**
## Context

Closes #119 
 
* add new capacity endpoints:
  * `/capacity/pools/:poolId` - Returns the current capacity for all products in a specific pool
  * `/capacity/pools/:poolId/products/:productId` - Returns the current capacity for a specific product in a specific pool
* add new `utilizationRate` field for all capacity endpoints

## Changes proposed in this pull request

* modified capacityEngine to take in filtering 3 params (poolId, productId, period). the behaviour is below:
  * no poolId / productId - returns capacity for ALL products across ALL pools
  * only productId - returns capacity for the specified product across ALL pools
  * only poolId - returns capacity for ALL products in the specified pool
  * poolId and productId - return capacity for the specified product in the specified pool
* clean up openapi docs by reusing CapacityResult component schema

NOTE: this PR is behind https://github.com/NexusMutual/cover-router/pull/135

## Test plan

* added unit tests

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [x] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [x] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: src/lib/capacityEngine.js [hunk 1]
```diff
@@ -7,6 +7,33 @@ const { selectAsset, selectProduct, selectProductPools } = require('../store/sel
...
+ * @param {BigNumber} capacityUsedNXM - The amount of capacity used in NXM.
+ * @returns {BigNumber} The utilization rate as a BigNumber. Returns undefined is capacity in NXM is missing
+ */
+function getUtilizationRate(capacityInAssets, capacityUsedNXM) {
+  const availableCapacityInNxm = capacityInAssets.find(asset => asset.assetId === 255)?.amount;
+  if (!availableCapacityInNxm || !capacityUsedNXM) {
+    return undefined;
+  }
+  const totalCapacity = availableCapacityInNxm.add(capacityUsedNXM);
+  return capacityUsedNXM.mul(10000).div(totalCapacity).toNumber() / 10000;
```
Comment (shark0der) [inline src/lib/capacityEngine.js:25]: jsdoc specifies this is a BigNumber, but you're returning a Number between 0 and 1 (eventually even more than 1 due to burns). I would suggest removing the `.toNumber() / 10000` part and returning that. Frontend will be able to convert that to percentage with 2 decimals.

Comment (shark0der) [reply src/lib/capacityEngine.js:25]: Secondly, did you test the edge case when `totalCapacity` is zero?

Comment (rackstar) [reply src/lib/capacityEngine.js:25]: * changed `utilizationRate` to be basis points (0-10,000)
* good catch on `totalCapacity` = 0, I'm not returning undefined.
  * it feels undefined is more appropriate in this edge case as 0 can be misinterpreted of having capacity but none is used

**Review by TeriThetha — APPROVED** [review]

**Review by TeriThetha — APPROVED** [review]

**Review by TeriThetha — APPROVED** [review]

**Review by TeriThetha — APPROVED** [review]

**Review by TeriThetha — APPROVED** [review]

File: src/lib/capacityEngine.js [hunk 2]
```diff
@@ -149,6 +236,7 @@ function capacityEngine(store, productIds, time, period = 30) {
         productId: Number(productId),
         availableCapacity: capacityInAssets,
         usedCapacity: capacityUsedNXM,
+        utilizationRate: getUtilizationRate(capacityInAssets, capacityUsedNXM),
```
Comment (MilGard91) [inline src/lib/capacityEngine.js:239]: nit: here we can just pass `capacityAvailableNXM` instead of `capacityInAssets` and we can skip filtering the array

File: src/lib/capacityEngine.js [hunk 3]
```diff
@@ -114,10 +199,12 @@ function capacityEngine(store, productIds, time, period = 30) {
...
```
Comment (MilGard91) [inline src/lib/capacityEngine.js:202]: nit: here we can just pass `capacityAvailableNXM` instead of `capacityInAssets` and we can skip filtering the array

**Review by MilGard91 — COMMENTED** [review]
Two small comments, beside that LGTM

File: src/lib/capacityEngine.js (hunk 2 above)
Comment (rackstar) [reply src/lib/capacityEngine.js:239]: good suggestion 👍
